File: backend/src/main/java/com/poshanforlife/api/dto/NotificationDto.java
package com.poshanforlife.api.dto;

import java.time.Instant;
import java.util.UUID;

public record NotificationDto(
        String id,
        String type,
        String title,
        String message,
        boolean read,
        String relatedEntityType,
        UUID relatedEntityId,
        Instant createdAt) {
}

File: android/app/src/main/java/com/poshanforlife/android/core/network/NotificationDto.kt
package com.poshanforlife.android.core.network

import kotlinx.serialization.Serializable

@Serializable
data class AppNotificationDto(
    val id: String,
    val type: String,
    val title: String,
    val message: String,
    val read: Boolean,
    val relatedEntityType: String? = null,
    val relatedEntityId: String? = null,
    val createdAt: String,
)

@Serializable
data class NotificationListResponseDto(
    val notifications: List<AppNotificationDto> = emptyList(),
    val unreadCount: Long = 0,
)
